kolibri/auth/serializers.py

```diff
@@ -25,7 +26,16 @@ def update(self, instance, validated_data):
             return super(BaseKolibriUserSerializer, self).update(instance, validated_data)
 
     def validate_username(self, value):
-        if FacilityUser.objects.filter(username__iexact=value).exists() | DeviceOwner.objects.filter(username__iexact=value).exists():
+        facility_user_query = FacilityUser.objects.filter(username__iexact=value)
+        device_owner_query = DeviceOwner.objects.filter(username__iexact=value)
+
+        user_id = self.initial_data.get("id", None)
+
+        if user_id:
+            facility_user_query = facility_user_query.exclude(id=user_id)
+            device_owner_query = device_owner_query.exclude(id=user_id)
+
+        if facility_user_query.exists() | device_owner_query.exists():
```

Comment:
Since these are different models, does `|` work properly, or would `or` be better? Actually, `.exists()`may return a `bool`, so this may be doing bitwise OR.

Reply:
I think a simple Python `or` is probably better here, I had just copied the code that was here already!